packages/utilities/src/setFocusVisibility.ts

```diff
@@ -14,22 +15,25 @@ export const IsFocusHiddenClassName = 'ms-Fabric--isFocusHidden';
  *
  * @param enabled - Whether to turn focus visibility on or off.
  * @param target - Optional target from which to get window in case no `providerElem` has been specified.
- * @param providerElem - Theme provider element to which the focus visibility classnames are attached. If no provider
- *                       element is passed in, the classnames are attached to the document body that contains `target`.
+ * @param registeredProviders - Array of provider refs that are associated with a FocusRectsProvider. If no array
+ *                              is passed in, the classnames are attached to the document body that contains `target`.
  */
-export function setFocusVisibility(enabled: boolean, target?: Element, providerElem?: Element): void {
-  let classList;
-  if (providerElem) {
-    classList = providerElem.classList;
+export function setFocusVisibility(
+  enabled: boolean,
+  target?: Element,
+  registeredProviders?: React.RefObject<HTMLElement>[],
+): void {
+  const updateClassList = (el: HTMLElement) => {
```

Comment:
This is almost a pure function and could easily be made so by taking `enabled` as an argument rather than getting it via the closure. This would allow us to define `updateClassList()` outside of `setFocusVisibility()` which avoids creating the function each time `setFocusVisibility()` is called. This isn't the biggest performance win ever but I know this code is perf-sensitive so every little bit helps imo.


```suggestion
  const updateClassList = (el: HTMLElement, enabled: boolean) => {
```

plus move `updateClassList()` outside of `setFocusVisibility()`.

Reply:
Good call. I've updated this to be a static function, and simplified `setFocusVisibility` a bit based on that and the getWindow change in your other comment.